Source/Noesis.Javascript/JavascriptContext.cpp

```diff
@@ -489,12 +494,19 @@ JavascriptContext::WrapObject(System::Object^ iObject)
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
-Handle<ObjectTemplate>
-JavascriptContext::GetObjectWrapperTemplate()
+Handle<FunctionTemplate>
+JavascriptContext::GetObjectWrapperConstructorTemplate(System::Type ^type)
 {
-	if (objectWrapperTemplate == NULL)
-		objectWrapperTemplate = new Persistent<ObjectTemplate>(isolate, JavascriptInterop::NewObjectWrapperTemplate());
-	return Local<ObjectTemplate>::New(isolate, *objectWrapperTemplate);
+    System::IntPtr ptrToConstructor;
+    if (!mTypeToConstructorMapping->TryGetValue(type, ptrToConstructor)) {
+        Local<FunctionTemplate> constructor = FunctionTemplate::New(GetCurrentIsolate());
+        JavascriptInterop::InitObjectWrapperTemplate(constructor->InstanceTemplate());
+        mTypeToConstructorMapping[type] = System::IntPtr(new Persistent<FunctionTemplate>(isolate, constructor));
+        Local<Context>::New(isolate, *mContext)->Global()->Set(isolate->GetCurrentContext(), String::NewFromUtf8(isolate, "Test"), constructor->GetFunction()); // TODO: do not do this by default

```

Comment:
this line must not be done here, but needs to be triggered from some kind of `RegisterConstructor`.